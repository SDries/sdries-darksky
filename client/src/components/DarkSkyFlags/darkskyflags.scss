@import '../../assets/styles/shared.scss';
@import '../../assets/styles/mixins.scss';

.DarkSkyFlags {
	margin-bottom: 40px;
	font-size: 1rem;
	line-height: 2rem;
	color: $red;
  border: 1px solid $red;
  @include border-radius(4px);
  padding: 20px;
}
