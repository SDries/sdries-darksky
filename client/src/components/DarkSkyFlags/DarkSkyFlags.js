import React, { Component } from 'react';
import './darkskyflags.scss';

class DarkSkyFlags extends Component {
  render() {
  	const flags = this.props.data;
    if(!flags) return null;

  	let unavailable;
  	if(flags['darksky-unavailable']) {
  		unavailable = <p>The Dark Sky data source supports the given location, however a temporary error has made the data unavailable.</p>;
  	}

  	if(!unavailable) return null;

    return (
      <div className="DarkSkyFlags">
      {unavailable}
      </div>
      );
  }
}

export default DarkSkyFlags;
