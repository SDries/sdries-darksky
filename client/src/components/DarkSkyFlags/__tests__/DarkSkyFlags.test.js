import React from 'react';
import DarkSkyFlags from '../DarkSkyFlags';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('react-darkskyflags', () => {
  test('renders without data', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<DarkSkyFlags />);
    const result = renderer.getRenderOutput();
    expect(result).toBeNull();
  });

  it('shallowly renders unavailable correctly', () => {
    const apiMock = {
      'darksky-unavailable': true
    };
    const renderer = new ShallowRenderer();
    renderer.render(<DarkSkyFlags data={apiMock} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
