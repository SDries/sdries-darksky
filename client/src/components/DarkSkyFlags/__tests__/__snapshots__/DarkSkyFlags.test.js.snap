// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-darkskyflags shallowly renders unavailable correctly 1`] = `
<div
  className="DarkSkyFlags"
>
  <p>
    The Dark Sky data source supports the given location, however a temporary error has made the data unavailable.
  </p>
</div>
`;
